tests/integration/targets/azure_rm_networkinterface/tasks/main.yml

```diff
@@ -319,6 +322,89 @@
       - output.state.ip_configurations[0].public_ip_address.name == "tn{{ rpfx }}"
       - output.state.enable_accelerated_networking
 
+- name: Create Application security group
+  azure_rm_applicationsecuritygroup:
+    resource_group: "{{ resource_group }}"
+    name: "{{ applicationsecuritygroup_name1 }}"
+
+- name: Create Application security group in secondary resource group
+  azure_rm_applicationsecuritygroup:
+    resource_group: "{{ resource_group_secondary }}"
+    name: "{{ applicationsecuritygroup_name2 }}"
+  register: asg
+
+- name: Create Nic with application security groups
+  azure_rm_networkinterface:
+    resource_group: "{{ resource_group }}"
+    name: "{{ nic_name1 }}"
+    virtual_network: "{{ vn.state.id }}"
+    subnet: "tn{{ rpfx }}"
+    create_with_security_group: True
+    public_ip: False
+    ip_configurations:
+      - name: ipconfig1
+        application_security_groups:
+          - "{{ applicationsecuritygroup_name1 }}"
+          - "{{ asg.id }}"
+        primary: True
+  register: output
+
+- name: assert creation succeeded
+  assert:
+    that:
+      - output.changed
+
+- name: Create Nic with application security groups (idempotent)
+  azure_rm_networkinterface:
+    resource_group: "{{ resource_group }}"
+    name: "{{ nic_name1 }}"
+    virtual_network: "{{ vn.state.id }}"
+    subnet: "tn{{ rpfx }}"
+    create_with_security_group: True
+    public_ip: False
+    ip_configurations:
+      - name: ipconfig1
+        application_security_groups:
+          - "{{ applicationsecuritygroup_name1 }}"
+          - "{{ asg.id }}"
```

Comment:
is application security groups order important? or it doesn't matter what is the order?
if order is not important then we may improve idempotence check.
but I guess we are ok.

Reply:
updated test